### PR TITLE
NDRS-1227: Use Genesis Validators When Fast Sync Joins Around Genesis

### PR DESCRIPTION
**Pull request by xcthulhu**
When joining right after genesis, fast sync will now use the genesis validators for finality signature verification.

### CODE REVIEW

File: node/src/components/linear_chain_sync/operations.rs [hunk 1]
```diff
@@ -113,6 +113,66 @@ where
...
+        {
+            return Ok(trusted_validator_weights.clone());
+        }
+        current_header_to_walk_back_from = fetch_and_store_block_header(
+            effect_builder,
+            *current_header_to_walk_back_from.parent_hash(),
+        )
+        .await?;
+    }
+}
```
Comment (afck) [inline node/src/components/linear_chain_sync/operations.rs:173]: Doesn't this duplicate [`Effect::get_era_validators`](https://github.com/casper-network/casper-node/blob/5dda5052bd1265de9164d7d3be45d24750686697/node/src/effect.rs#L1557), except that that retrieves the validators from the global state? But at that point, we should have the global state available so it should work, shouldn't it?

Comment (xcthulhu) [reply node/src/components/linear_chain_sync/operations.rs:173]: Almost, but not exactly.  `Effect::get_era_validators` can does storage and contract-runtime requests.  We don't have anything in either storage or the state trie at the point this code is called.  The functions are similar, however: they both fallback to the chainspec in the special case where we are asking for the validators for era 0.

File: node/src/components/linear_chain_sync/operations.rs [hunk 2]
```diff
@@ -113,6 +113,66 @@ where
...
+        let genesis_validator_weights: BTreeMap<PublicKey, U512> = chainspec
+            .network_config
+            .accounts_config
+            .accounts()
+            .iter()
+            .filter_map(|account_config| {
+                if account_config.is_genesis_validator() {
+                    Some((
+                        account_config.public_key(),
+                        account_config.bonded_amount().value(),
```
Comment (afck) [inline node/src/components/linear_chain_sync/operations.rs:137]: Does that include delegators? (Not sure how that works at genesis.)

Comment (xcthulhu) [reply node/src/components/linear_chain_sync/operations.rs:137]: I _think_ so?  Paging @Fraser999 ... maybe we should have a function that shares this logic in both the genesis code and here as well.

**Pull request merged**